dynamics/utils.py

```diff
@@ -188,10 +193,23 @@ def clear(self) -> None:
         self.con.execute(self._clear_sql)
 
 
-try:
-    from django.core.cache import cache
-except ImportError:
-    cache = SQLiteCache()
+class LocalCache:
```

Comment:
Isn't this class interface unnecessary? The cache import from django or the custom sqlite cache gets called the first time dynamics/utils.py is imported, after which `dynamics.utils.cache` is the same cache instance no matter where it is imported.

Reply:
My motivation for this change is to give the user some level of control over the timing of the creation of this cache object; currently, it occurs at module import time.

In the above scenario, cache is loaded or created when the user first requests it, and as you say, the same item is used thereafter.

Later, this pattern can further carve out sync vs. async scenarios (since interacting with a disk cache is also I/O which ought to be awaited in async contexts).

Reply:
Ok, makes sense 👍 Could you rename this class to `CacheSingleton` or something similar? When used with Django's cache, the cache might not be "local".

Reply:
I named it singleton so we can use it for items other than cache if required down the line.

Reply:
Ok, seems like a good idea. Can we name the class plural `Singletons` and indicate in the class docstring that it is an interface for all kinds of singleton objects?